File: 1321-get-equal-substrings-within-budget/get-equal-substrings-within-budget.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <string.h>

int equalSubstring(char *s, char *t, int maxCost) {
    int n = strlen(s);
    int res = 0, l = 0, cost = 0;

    for (int r = 0; r < n; r++) {
        cost += abs(s[r] - t[r]);
        while (cost > maxCost) {
            cost -= abs(s[l] - t[l]);
            l++;
        }
        res = fmax(res, r - l + 1);
    }

    return res;
}

File: 1321-get-equal-substrings-within-budget/get-equal-substrings-within-budget.py
class Solution(object):
    def equalSubstring(self, s, t, maxCost):
        n = len(s)
        res, l, cost = 0, 0, 0

        for r in range(n):
            cost += abs(ord(s[r]) - ord(t[r]))
            while cost > maxCost:
                cost -= abs(ord(s[l]) - ord(t[l]))
                l += 1
            res = max(res, r - l + 1)
        
        return res

        